src/NHibernate/ObjectNotFoundByUniqueKeyException.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using NHibernate.Impl;
+
+namespace NHibernate
+{
+	/// <summary>
+	/// Thrown when entity can't be found by given unique key
+	/// </summary>
+	[Serializable]
+	public class ObjectNotFoundByUniqueKeyException : HibernateException
```

Comment:
It is not enough for an exception class to have the `[Serializable]` attribute. The `Exception` base class implements `ISerializable`, so, when new members are added in a derived class, `GetObjectData` must be overridden for serializing them. And in any case, the deserialization constructor must be re-declared.
See `NHibernate.ADOException` for an example.

Reply:
Fixed